Sources/PlaydateKit/Core/Sound.swift

```diff
@@ -44,6 +50,37 @@ public enum Sound {
         @discardableResult public func play(repeat: Int32 = 1) -> Int32 {
             fileplayer.play.unsafelyUnwrapped(pointer, `repeat`)
         }
+        
+        private struct CallbackData {
+            var callback: (() -> Void)?
+        }
+        private var _callbackData: UnsafeMutablePointer<CallbackData>?
+
+        /// Installs a closure that will be called when playback has completed. This is implemented
+        /// by wrapping the closure in a struct that is stored in a dynamically allocated block, which
+        /// is passed as `userData` to `setFinishCallback` below.  
```

Comment:
I don't think the implementation details should go in the doc comments here as they aren't really relevant to users of the library, but might be nice to leave in a regular comment.

Reply:
See [1f2d7b4](https://github.com/finnvoor/PlaydateKit/pull/46/commits/1f2d7b4b8597c07d1613f2df00de06b073c03770)